### PR TITLE
CIRC-1958: Handle circulation rules update events

### PR DESCRIPTION
**Pull request by OleksandrVidinieiev**
## Purpose
Remove timer-based circulation rules cache updates and consume events published by circulation-storage instead.
[CIRC-1958](https://issues.folio.org/browse/CIRC-1958)

## Approach
- add integration with Kafka
- subscribe to circulation rules update events
- update circulation rules cache with rules received in the event payload
- place each consumer in a separate consumer group so that each instance of mod-circulation receives the event, otherwise the cache will only be updated in a single instance 

#### TODOS and Open Questions
- [ ] Check logging

### CODE REVIEW

File: src/main/java/org/folio/circulation/EventConsumerVerticle.java [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+      .onFailure(t -> log.error("stop:: verticle stop failed", t))
+      .onComplete(promise);
+  }
+
+  private Future<Void> stopConsumers() {
+    return Future.all(
+      consumers.stream()
+        .map(KafkaConsumerWrapper::stop)
+        .toList())
+      .onSuccess(v -> log.info("stop:: event consumers stopped"))
```
Comment (roman-barannyk) [inline src/main/java/org/folio/circulation/EventConsumerVerticle.java:80]:
```suggestion
      .onSuccess(v -> log.info("stopConsumers:: event consumers stopped"))
```

File: src/main/java/org/folio/circulation/EventConsumerVerticle.java [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+      .onFailure(t -> log.error("stop:: failed to stop event consumers", t))
```
Comment (roman-barannyk) [inline src/main/java/org/folio/circulation/EventConsumerVerticle.java:81]:
```suggestion
      .onFailure(t -> log.error("stopConsumers:: failed to stop event consumers", t))
```

File: src/main/java/org/folio/circulation/EventConsumerVerticle.java [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+
+  private static SubscriptionDefinition buildSubscriptionDefinition(DomainEventType eventType) {
+    return SubscriptionDefinition.builder()
+      .eventType(eventType.name())
+      .subscriptionPattern(eventType.getKafkaTopic().fullTopicName(TENANT_ID_PATTERN))
+      .build();
+  }
+
+  private KafkaConfig buildKafkaConfig() {
+    log.info("getKafkaConfig:: building Kafka config");
```
Comment (roman-barannyk) [inline src/main/java/org/folio/circulation/EventConsumerVerticle.java:118]:
```suggestion
    log.info("buildKafkaConfig:: building Kafka config");
```

File: src/main/java/org/folio/circulation/EventConsumerVerticle.java [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+    KafkaConfig config = KafkaConfig.builder()
+      .envId(vertxConfig.getString(KAFKA_ENV))
+      .kafkaHost(vertxConfig.getString(KAFKA_HOST))
+      .kafkaPort(vertxConfig.getString(KAFKA_PORT))
+      .okapiUrl(vertxConfig.getString(OKAPI_URL))
+      .replicationFactor(Integer.parseInt(vertxConfig.getString(KAFKA_REPLICATION_FACTOR, "1")))
+      .maxRequestSize(Integer.parseInt(vertxConfig.getString(KAFKA_MAX_REQUEST_SIZE, "10")))
+      .build();
+
+    log.info("getKafkaConfig:: {}", config);
```
Comment (roman-barannyk) [inline src/main/java/org/folio/circulation/EventConsumerVerticle.java:130]:
```suggestion
    log.info("buildKafkaConfig:: {}", config);
```

File: src/main/resources/log4j2.properties [hunk 5]
```diff
@@ -5,7 +5,7 @@ packages = org.folio.okapi.common.logging
 filters = threshold
 
 filter.threshold.type = ThresholdFilter
-filter.threshold.level = info
+filter.threshold.level = debug
```
Comment (OleksandrVidinieiev) [inline src/main/resources/log4j2.properties:8]: Revert

**Review by alexanderkurash — APPROVED** [review]

**Review by roman-barannyk — APPROVED** [review]

**Pull request merged**